crates/rpc/src/method/subscribe_events.rs

```diff
@@ -31,7 +31,8 @@ impl Params {
             }
         }
         if let Some(keys) = &self.keys {
-            if keys.is_empty() {
+            let no_key_constraints = keys.iter().flatten().count() == 0;
```

Comment:
BTW: we could probably move this computation to a constructor of this type instead? Otherwise we should really check how expensive this calculation is.